controllers/activemqartemis_reconciler.go

```diff
@@ -1633,13 +1631,12 @@ func (reconciler *ActiveMQArtemisReconcilerImpl) NewPodTemplateSpecForCR(customR
 	clog.Info("Total volumes ", "volumes", podSpec.Volumes)
 
 	// this depends on init container passing --java-opts to artemis create via launch.sh *and* it
-	// not getting munged on the way.
-	// REVISIT: should an existing value be respected here, only when we expose JAVA_OPTS in the CR?
+	// not getting munged on the way. We CreateOrAppend to any value from spec.Env
 	javaOpts := corev1.EnvVar{
 		Name:  "JAVA_OPTS",
 		Value: "-Dbroker.properties=/amq/extra/configmaps/" + brokerPropertiesConfigMapName + "/broker.properties",
 	}
-	environments.Create(podSpec.InitContainers, &javaOpts)
+	environments.CreateOrAppend(podSpec.InitContainers, &javaOpts)
```

Comment:
as final user maybe I would expect see my env vars with the same value defined in the CR
as operator maybe I should avoid to change my env vars
a solution could be that the operator ignore internal vars as JAVA_OPTS and include a prefix/suffix env var in the internal env vars. i.e. JAVA_OPTS="$JAVA_OPTS_PRE ... $JAVA_OPTS_SUF"
so users are not able to set JAVA_OPTS but they are able to set $JAVA_OPTS_PRE or $JAVA_OPTS_SUF

Reply:
I think JAVA_OPTS is some what of a special case b/c it is interpreted by the init container, artemis create command and is used for brokerProperties. I think we can in the future configure the brokerProperties url in some better way and then not have to treat is specially w.r.t append. 
But for sure I don't want to encourage the use of env JAVA_OPTS. it is particular to the artemis launch scripts... way too low level. Ading PRE POST special cases would make it worse imho.
To configure the JVM, use JDK_JAVA_OPTIONS